src/mgm.rs

```diff
@@ -165,11 +292,11 @@ impl MgmKey {
             e
         })?;
 
-        if item.len() != DES_LEN_3DES {
+        if item.len() != C::KEY_SIZE as usize {
```

Comment:
Bug: 3DES and AES-192 have the same key size, so this API will let the user read a protected 3DES key as an AES-192 key, and vice versa. The Yubikey is aware of the difference, because `Ins::SetMgmKey` takes `new_key.algorithm_id()` as an argument.

As I noted in https://github.com/iqlusioninc/yubikey.rs/issues/330#issuecomment-2564689714, we need to figure out how to distinguish a 3DES management key from an AES-192 management key. I'm going to hunt around in the `ykman` source to see how they do it.

Reply:
Aha, we can (for firmware 5.3 and up, where it is available) just use the GET METADATA extension.